### PR TITLE
feat: migrate from dart_skills_lint to skills_lint package

### PR DESCRIPTION
**Pull request by reidbaker**
This pr was agent authored but it modifies a workflow and reidbaker-agent does not have workflow scope.
The skills_lint code is moving out of this repo to its own github repo github.com/google/skills_lint.dart.
While working the agent found a duplicate test code so I removed that as well. 
- @reidbaker 

---
Agent authored description 
Migrates from the legacy `dart_skills_lint` package to the standalone `skills_lint` package hosted in `google/skills_lint.dart`, updating configurations, test imports, workspace definitions, and CI path triggers.

### CODE REVIEW

File: .github/workflows/skills_tool.yaml [hunk 1]
```diff
@@ -7,11 +7,17 @@ on:
   pull_request:
     paths:
       - '.github/workflows/skills_tool.yaml'
+      - 'pubspec.yaml'
+      - 'resources/**'
```
Comment (reidbaker) [inline .github/workflows/skills_tool.yaml]: I dont think resources makes sense as a trigger hear.

Comment (reidbaker) [reply .github/workflows/skills_tool.yaml]:
```suggestion
```

File: .github/workflows/skills_tool.yaml [hunk 2]
```diff
@@ -7,11 +7,16 @@ on:
...
       - '.github/workflows/skills_tool.yaml'
+      - 'pubspec.yaml'
+      - 'skills/**'
       - 'tool/generator/**'
   push:
     branches: [ main ]
     paths:
       - '.github/workflows/skills_tool.yaml'
+      - 'pubspec.yaml'
+      - 'resources/**'
```
Comment (reidbaker) [inline .github/workflows/skills_tool.yaml]:
```suggestion
```

File: tool/generator/test/custom_skill_rules/last_modified_rule.dart [hunk 3]
```diff
@@ -14,7 +14,17 @@ class LastModifiedRule extends SkillRule {
   Future<List<ValidationError>> validate(SkillContext context) async {
     final errors = <ValidationError>[];
     final yaml = context.parsedYaml;
-    if (yaml == null) return errors;
+    if (yaml == null) {
```
Comment (reidbaker) [inline tool/generator/test/custom_skill_rules/last_modified_rule.dart:17]: Intentional change to make this rule better handle edge cases. It should not pass on a malformed frontmatter.

File: tool/generator/test/lint_skills_test.dart
Comment (reidbaker) [inline tool/generator/test/lint_skills_test.dart:1]: There is some annoying directory manipulation in this test. I filed https://github.com/google/skills_lint.dart/issues/2 to make it better in the future and the naive solution failed if you ran the test from the root instead of from the generator directory.

File: tool/generator/test/validate_skills_test.dart [hunk 4]
```diff
@@ -444,10 +444,6 @@ void main() {
         await runner.run(['validate-skill', configFile.path]);
       }, getCurrentDirectory: () => tempDir);
 
-      expect(
```
Comment (reidbaker) [inline tool/generator/test/validate_skills_test.dart:447]: This was a duplicate line. I also think there are duplicate tests but decided to keep my change in this file to only the duplicate validation.

**Review by johnpryan — APPROVED** [review]
